website/events/models/event.py

```diff
@@ -181,6 +181,8 @@ class Event(models.Model, metaclass=ModelTranslateMeta):
 
     tpay_allowed = models.BooleanField(_("Allow Thalia Pay"), default=True)
 
+    shift = models.OneToOneField("sales.Shift", models.SET_NULL, null=True)
```

Comment:
I do not completely understand why the shift is part of the event and not the other way around, I don't see the shift used anywhere in the events package. Which means that this is a dependency from events on sales that does not seem necessary to me. Wouldn't it then be better to make the field a part of the shift instead?

Reply:
For me it feels fundamentally weird to have a dependency from sales on events... Shifts should be standalone things if you ask me. It's only that certain events will have a shift.

Reply:
Yes that is also true, but you create the optional dependency on events already by checking for the event attribute. So now you have a dependency in two places I think?

I'm not sure, maybe it would just be better to not set the title using the event? That is the only thing it does right?
I am also afraid that people will not set the shift on the event just because they are lazy, since shifts will probably only be created once they will need one.

Reply:
It's not a _real_ dependency as I use the `has_attr` thingy. But you are right, maybe I should just delete it. Or at least, delete the title property. I think we could keep the field in the events. In the end, I think it will become something similar to pizza events.

Reply:
If you're not using it then it should probably not exist.

To be fair though, pizzas and events are pretty tightly coupled (which is bad). They also have a circular dependency. But at least the key to the Event is in the PizzaEvent :p

Reply:
Maybe in (far) the future, `sales` could replace `pizzas` 🤔...

I will remove the title thingy from the sales model, but I do want to keep the the OneToOneField actually so we can just start keeping the data about Shifts related to Events from the beginning, just like we're doing with Photo albums as of now.

Reply:
You make the assumption that people are going to fill in a field that is not required. :)

I am not sure if you were already in the Technicie when we introduced categories for events, but the default value was always set to 'Other'. And thus people did not change the value of the field to the correct one. I'm fine with keeping the field but I expect the same to happen here.

Reply:
We can see, a man can dream. I do have faith, as the 'category' field doesn't enable a real feature or something, while this will (I could even think about writing the code in such a way that committee members can only create shifts that are linked to an event - but not in this PR).

Reply:
But you should not want shifts that are only connected to events. Maybe you want shifts without events too.

And for that the field has to be in the shift since it has to exist before you can connect it to and event.

Reply:
I don't mean implementing it at the model level. I mean disabling the add permission on the ShiftAdmin for members that don't have the shift manager override permission, and only enabling the add permission via the EventAdmin for them.